Sources/MongoDBVapor/MongoDBVapor.swift

```diff
@@ -1 +1,65 @@
-private func doSomething() {}
+import MongoSwift
+import Vapor
+
+extension Application {
+    /// A global `MongoClient` for use throughout the application. The client is thread-safe
+    /// and backed by a pool of connections so it should be shared across event loops.
+    public var mongoClient: MongoClient {
+        get {
+            self.storage[MongoClientKey.self]!
+        }
+        set {
+            self.storage[MongoClientKey.self] = newValue
+        }
+    }
+
+    private struct MongoClientKey: StorageKey {
+        // swiftlint:disable nesting
+        typealias Value = MongoClient
+    }
+
+    /**
+     * Configures a global `MongoClient` with the specified options. The client may then be accessed via the
+     * `mongoClient` computed property on the `Application`. The client will use the `Application`'s `EventLoopGroup`
+     * for executing operations.
+     *
+     * For example:
+     * ````
+     *  let app = Application() // a Vapor Application
+     *  try app.configureMongoDB() // configures a MongoClient
+     *  app.mongoClient.listDatabases() // a client is now accessable via the Application
+     * ````
+     *
+     * - Parameters:
+     *   - connectionString: the connection string to connect to.
+     *   - options: optional `MongoClientOptions` to use for the client.
+     *
+     * - Throws:
+     *   - A `MongoError.InvalidArgumentError` if the connection string passed in is improperly formatted.
+     */
+    public func configureMongoDB(
+        _ connectionString: String = "mongodb://localhost:27017",
+        options: MongoClientOptions? = nil
+    ) throws {
+        let client = try MongoClient(connectionString, using: self.eventLoopGroup, options: options)
+        self.mongoClient = client
+    }
+
+    /**
+     * Handles MongoDB related cleanup. Call this method when shutting down your application.
+     * If an error occurs while closing the client, it will be logged via the `Application`'s logger.
+     * For example:
+     * ````
+     *  try app.cleanupMongoDB()
+     *  app.shutdown()
+     * ````
+     */
+    public func cleanupMongoDB() {
+        do {
+            try self.mongoClient.syncClose()
+        } catch {
+            self.logger.error("Failed to shut down MongoDB client: \(error)")
+        }
+        cleanupMongoSwift()
```

Comment:
@0xTim a question: is having multiple `Application`s running at once (like via the same `Run` executable) something Vapor users would ever do? Or is it fairly safe to assume a 1:1 relationship?

This `cleanupMongoSwift` method is an unfortunate artifact of our wrapping the C driver which has a global cleanup method `mongoc_cleanup` we're wrapping here: http://mongoc.org/libmongoc/1.16.2/mongoc_cleanup.html 

Basically it frees some global state and after its called the driver is no longer usable. So if people use multiple `Application`s we could run into an issue here if only one has been shut down and others are still trying to use the driver.

Reply:
Anyone running multiple `Application`s in the same process is definitely an edge case. I'm not sure Vapor would even allow it. So you can probably safely assume it's fine for now

Reply:
Do you mean each test having a separate application? That should be supported and fine unless you have any global static config 🤔 But adding in the docs to tell them to shut it down should be fine

Reply:
yeah there is some global stuff the C driver is doing unfortunately